app/src/main/java/com/sun/makeupwindow/data/source/local/dao/FavoriteProductDaoImpl.kt

```diff
@@ -0,0 +1,63 @@
+package com.sun.makeupwindow.data.source.local.dao
+
+import android.content.ContentValues
+import com.sun.makeupwindow.data.model.Product
+import com.sun.makeupwindow.data.source.local.db.AppDatabase
+
+class FavoriteProductDaoImpl private constructor(
+    private val appDatabase: AppDatabase
+) : FavoriteProductDao {
+
+    private val writableDatabase = appDatabase.writableDatabase
+    private val readableDatabase = appDatabase.readableDatabase
+
+    override fun getFavoriteProduct(): List<Product> {
+        val listProduct = mutableListOf<Product>()
+        val query =
+            "SELECT * FROM ${Product.TABLE_FAVORITE_NAME} INNER JOIN ${Product.TABLE_NAME} " +
```

Comment:
```suggestion
        val query = "SELECT * FROM ${Product.TABLE_FAVORITE_NAME} INNER JOIN ${Product.TABLE_NAME} " +
```
tách INNER JOIN ra 1 string riêng luôn